DAGE-65: Analyze MTEB results

Intrinsical-AI opened the pull request:
TASK: [DAGE-65](https://sease.atlassian.net/browse/DAGE-65?atlOrigin=eyJpIjoiNzQ4MWIxYmJjNzQ5NGU4MDlmYjBjMDk5ZmU3NTVmMTciLCJwIjoiaiJ9)

- Updated main
- Executed NFCorpus (medical dataset) + Mini-L6-v2 (lightweight model)

Check the run artifacts and analysis: [JIRA-COMMENT](https://sease.atlassian.net/browse/DAGE-65?focusedCommentId=18287)

Intrinsical-AI committed:
DAGE-65: In-memory MTEB pipeline entrypoint (cached embeddings, dynamic task build, dataset/split via Config)

Intrinsical-AI committed:
DAGE-65: Config – add dataset_name/split, JSONL validation, and default output dir

Intrinsical-AI committed:
DAGE-65: Add example nfcorpus config (retrieval)

Intrinsical-AI committed:
(clean) DAGE-65: polished comments - code styling

Intrinsical-AI committed:
config: provide default dataset_name ('custom-dataset') and split ('test') for backward compatibility

Intrinsical-AI committed:
added script to download a beir dataset

Intrinsical-AI committed:
MTEB script

Intrinsical-AI committed:
MTEB dataset crafting script

nicolo-rinaldi commented on rre-tools/embedding-model-evaluator/README.md:
````diff
@@ -1,5 +1,44 @@
+# Run exact search evaluator with  yaml config file
+uv run embedding-model-evaluator --config "config.yaml"
+```
+> "mteb_retrieval_dataset_generator" CLI params:
````
What is this line about?

Intrinsical-AI replied:
This was supposed to be a descriptive list of params / usage for the M2 that I forgot to add. Thx, comming in the next commit!

nicolo-rinaldi commented on rre-tools/embedding-model-evaluator/src/embedding_model_evaluator/main.py:
```diff
@@ -1,71 +1,129 @@
+    # --- Task instance (in-memory) ---
+    try:
+        task = _build_task(
+            task_key=config.task_to_evaluate,
```
you are passing "retrieval" right? Shouldn't you pass "CustomRetrievalTask"?

Intrinsical-AI replied:
(at the begining of the file:)

```
TASKS_NAME_MAPPING = {
    "retrieval": "CustomRetrievalTask",
    "reranking": "CustomRerankingTask",
}
```
(in _build_task:)

`task_cls_name = TASKS_NAME_MAPPING.get(task_key, "CustomRetrievalTask")`

Then should be
`task_cls_name = "CustomRetrievalTask"` when `task_key = "retrieval"`.

nicolo-rinaldi commented on rre-tools/embedding-model-evaluator/scripts/mteb_retrieval_dataset_generator.py:
```diff
@@ -0,0 +1,451 @@
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+
+def s(v: Any) -> str:
```
Can we call this something like `to_string()`?

Intrinsical-AI replied:
This shouldn't be called 's'. I did not payed much attention to the redability. I will upgrade the script naming in general. Thank you

nicolo-rinaldi commented on rre-tools/embedding-model-evaluator/scripts/mteb_retrieval_dataset_generator.py:
```diff
@@ -0,0 +1,451 @@
+    pos_docs_by_q: Dict[str, Set[str]] = {}
+    neg_count_by_q: Dict[str, int] = {}
+    pairs_seen: Set[Tuple[str, str]] = set()
+    qrels_total = cand_total = cand_pos = cand_neg = 0
```
typing is missing

Intrinsical-AI replied:
I think neither mypy or ruff tag it as problematic. Wonder why. Anyway: for constency, cleansing and redability I agree 👍

Intrinsical-AI replied:
I just pushed this also (was missing). Thanks again

Intrinsical-AI committed:
feat(embedding-evaluator): add script for MTEB retrieval dataset download and processing

Offline evaluation with MTEB on custom retrieval tasks support. Following Nicolos feedback, improved naming and overall readability.

Intrinsical-AI committed:
feat(embedding-evaluator): add script for MTEB retrieval dataset download and processing

Offline evaluation with MTEB on custom retrieval tasks support. Following Nicolos feedback, improved naming and overall readability.

Intrinsical-AI committed:
feat(embedding-evaluator): add script for MTEB retrieval dataset download and processing

Following Nicolo's feedback: strict typing

nicolo-rinaldi reviewed (approved):
Approving since you are addressing the comments

Intrinsical-AI committed:
clean(embedding-model-evaluator/tests/unit): renamed tests to match current __expects__ pattern

nseidan commented on rre-tools/embedding-model-evaluator/tests/unit/test_mteb_config.py:
```diff
@@ -12,7 +12,7 @@ def config() -> Config:
     return Config.load("embedding-model-evaluator/tests/unit/resources/valid_config.yaml")
 
 
-def test_valid_config_expect_all_params_read(config: Config) -> None:
```
what is wrong with the original naming?

Intrinsical-AI replied:
We're using unified test naming pattern with `__expects__`

nseidan commented on rre-tools/embedding-model-evaluator/tests/unit/test_custom_tasks.py:
```diff
@@ -78,7 +78,7 @@ def test_custom_retrieval_task_expect_valid_data_shape(tmp_path: Path) -> None:
 
 @pytest.mark.filterwarnings("ignore::DeprecationWarning")
-def test_custom_reranking_task_expect_valid_data_shape(tmp_path: Path) -> None:
+def test_reranking_task_with_valid_data__expects__loads_dataset_with_positive_and_negative_examples(tmp_path: Path) -> None:
```
better to keep `custom_reranking`

Intrinsical-AI replied:
Ok, I will add custom

nseidan commented on rre-tools/embedding-model-evaluator/config.yaml:
```diff
@@ -2,16 +2,16 @@
 
 # Corpus jsonl file path with records - <id,title,text>.
-corpus_path: "resources/data/corpus.jsonl"
+corpus_path: "resources/mteb_datasets/scifact/test/corpus.jsonl"
```
what is `scifact` ?

Intrinsical-AI replied:
A general science dataset used for IR in BEIR and MTEB

Intrinsical-AI committed:
(feedback-improvements:) test naming aligment, unified to nfcorpus as default in the script, tests, README, and config file

Intrinsical-AI committed:
Merge branch 'dataset-generator' into DAGE-65-clean
